llmfoundry/models/mpt/modeling_mpt.py

```diff
@@ -826,7 +826,8 @@ def set_output_embeddings(
             self.transformer.set_input_embeddings(new_embeddings)
 
     def tie_weights(self) -> None:
-        self.lm_head = None
+        if getattr(self.config, "tie_word_embeddings", True):
```

Comment:
Can you add a really simple test that just checks that `tie_weights` does as expected if config is true, and no op if config is false?

Reply:
looks there is a test already for this in `tests/models/test_model.py::test_generate_with_device_map`

Reply:
I meant, an explicit test that just calls `tie_weights` and makes sure it behaves as we want in both cases (also CI is failing, need to run precommit)

Reply:
done